### PR TITLE
CI: Check for inconsistent pandas namespace usage

### PR DESCRIPTION
**Pull request by dsaxton**
Adding a CI check that we aren't (for instance) using Series(...) and pd.Series(...) in the same file. ~This is kept intentionally small in scope (checking only DataFrame and Series for one file name right now) since this is very common in the code base and I'm not sure if this is something we'd actually want to enforce.~

### CODE REVIEW

File: ci/code_checks.sh [hunk 1]
```diff
@@ -252,6 +252,12 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     MSG='Check code for instances of os.remove' ; echo $MSG
     invgrep -R --include="*.py*" --exclude "common.py" --exclude "test_writers.py" --exclude "test_store.py" -E "os\.remove" pandas/tests/
     RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    MSG='Check for inconsistent use of pandas namespace' ; echo $MSG
+    grep -R -l --include="test_missing.py" " Series(" | xargs grep -n "pd\.Series("; test $? -gt 0
+    RET=$(($RET + $?))
+    grep -R -l --include="test_missing.py" " DataFrame(" | xargs grep -n "pd\.DataFrame("; test $? -gt 0
```
Comment (dsaxton) [inline ci/code_checks.sh:259]: Will put this in a function if we want to check for this. I didn't see an obvious way to use invgrep for this check so writing a custom command instead.

File: ci/code_checks.sh [hunk 2]
```diff
@@ -38,6 +38,11 @@ function invgrep {
     return $((! $EXIT_STATUS))
 }
 
+function check_namespace {
+    grep -R -l --include="test_analytics.py" " ${1}(" | xargs grep -n "pd\.${1}("
```
Comment (jreback) [inline ci/code_checks.sh:42]: make the includes another variable so it can be (slowly) expanded over time

**Review by jreback — CHANGES_REQUESTED** [review]
yeah we had an issue where we talked about this a while back . There was an analysis and IIRC we were basically split between these usages. The discussion basically said that an individual file should be consistent (either pd.* or just Series/DataFrame), but we wouldn't do the effort to clean everything up.

@jorisvandenbossche was for not clean
me and @jbrockmendel though we should just standarize on Series/DataFrame

IIRC.

I am still in favor of using Series/DataFrame everywhere in tests (and not pd.*)

**Comment by jbrockmendel** [comment]
I agree with @jreback: small preference towards not `pd.`, preference for within-file consistency, not worth bikeshedding on this topic

File: ci/code_checks.sh [hunk 3]
```diff
@@ -38,6 +38,11 @@ function invgrep {
     return $((! $EXIT_STATUS))
 }
 
+function check_namespace {
+    grep -R -l --include="${2}" --exclude="${3}" " ${1}(" | xargs grep -n "pd\.${1}("
```
Comment (jreback) [inline ci/code_checks.sh:42]: can you use named parameters here, e.g. something like

```
local -r INCLUDE="${1}"
```

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
yeah it think this is ok to add. will have to figure out the best way to expand this coverage. maybe a file for includes with the test files to check? (or a config like file).

**Comment by dsaxton** [comment]
> yeah it think this is ok to add. will have to figure out the best way to expand this coverage. maybe a file for includes with the test files to check? (or a config like file).

The diffs will be very large but it may be easier just to fix this globally for the different classes. I did this for Series just now (can revert if the PR is too large) and could do for DataFrame, etc. if worthwhile.

**Comment by jreback** [comment]
> > yeah it think this is ok to add. will have to figure out the best way to expand this coverage. maybe a file for includes with the test files to check? (or a config like file).
> 
> The diffs will be very large but it may be easier just to fix this globally for the different classes. I did this for Series just now (can revert if the PR is too large) and could do for DataFrame, etc. if worthwhile.

oh i think this is fine (yeah do DataFrame separately); only issue is have to make sure all passing (in case we are missing imports). ci is a bit cranky now.

**Comment by jreback** [comment]
and actually merge master once again (as just merged big PR). ping when you are ready.

**Comment by dsaxton** [comment]
> and actually merge master once again (as just merged big PR). ping when you are ready.

Looks to be mostly green (only annoying thing that can happen are false positives due to Series(...) being used in comments but not code, so I changed those as well)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton

**Pull request merged**